token/link/github.go

```diff
@@ -0,0 +1,41 @@
+package link
+
+import (
+	"strings"
+
+	"github.com/fabric8-services/fabric8-auth/client"
+
+	"github.com/satori/go.uuid"
+	"golang.org/x/oauth2"
+	"golang.org/x/oauth2/github"
+)
+
+const (
+	gitHubProviderID = "2f6b7176-8f4b-4204-962d-606033275397" // Do not change! This ID is used as provider ID in the external token table
+)
+
+type GitHubConfig struct {
```

Comment:
Could you add a "Type" string which would be set to "git"  - we would need the provider type string when we return the header 
```
WWW-Authenticate: Link provider=git, url=https://auth.openshift.io/api/token/link?redirect=https%3A%2F%2Fopenshift.io?scope=user%20public_repo, description=”Git token is missing. Link Git account”
```

Reply:
Sure, we can add a type string here. But it's rather a github type because different git providers will have different types: github, gitlab, etc.
But why would you need a provider type in WWW-Authenticate:? How it would help?

Reply:
I would need the type string to generate this message : 

"WWW-Authenticate: Link provider=github, url=https://auth.openshift.io/api/token/link?redirect=https%3A%2F%2Fopenshift.io?scope=user%20public_repo, description=”Github token is missing. Link Github account”

All the places were we use "github" would be replaced by "openshift" , which is going away but still might need in future.

Reply:
But how the provider type name would help the client? Why do you want to include it in the header?

Reply:
I added provider type name - https://github.com/fabric8-services/fabric8-auth/pull/136/commits/83df82b919fa362f0da659472914802d6bb6f96c